### PR TITLE
Added missing GetInstance to providers

### PR DESCRIPTION
**Pull request by arknable**
1. Added missing `GetInstance` method to some providers
2. Renamed existing `GetInstanceByID` methods as `GetInstance`

### CODE REVIEW

File: digitalocean/digital_ocean_instance.go [hunk 1]
```diff
@@ -54,25 +54,20 @@ func (do *DigitalOcean) CreateInstance(ctx *lepton.Context) error {
 	return nil
 }
 
-// GetInstanceByID returns the instance with the id passed by argument if it exists
-func (do *DigitalOcean) GetInstanceByID(ctx *lepton.Context, id string) (droplet *lepton.CloudInstance, err error) {
+// GetInstance returns instance with given name
+func (do *DigitalOcean) GetInstance(ctx *lepton.Context, name string) (*lepton.CloudInstance, error) {
```
Comment (fabioDMFerreira) [inline digitalocean/digital_ocean_instance.go:58]: Heads up that having named returned arguments simplifies the function. Anyway, I think this is a matter of code style.

File: lepton/provider.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ type Provider interface {
 	CreateInstance(ctx *Context) error
 	ListInstances(ctx *Context) error
 	GetInstances(ctx *Context) ([]CloudInstance, error)
-	GetInstanceByID(ctx *Context, id string) (*CloudInstance, error)
+	GetInstance(ctx *Context, name string) (*CloudInstance, error)
```
Comment (fabioDMFerreira) [inline lepton/provider.go:39]: I've some concerns about the name of the function `GetInstance`. This name is too similar to `GetInstances`. Having a name like `GetInstancebyName` increases the code readability.

File: aws/aws_instance.go [hunk 3]
```diff
@@ -319,6 +319,16 @@ func (p *AWS) GetInstances(ctx *lepton.Context) ([]lepton.CloudInstance, error)
...
 }
 
+// GetInstance returns instance with given name
+func (p *AWS) GetInstance(ctx *lepton.Context, name string) (*lepton.CloudInstance, error) {
+	for _, instance := range getAWSInstances(ctx.Config().CloudConfig.Zone, nil) {
+		if instance.Name == name {
+			return &instance, nil
+		}
+	}
+	return nil, nil
```
[the remark this answers is not in the document]
Comment (fabioDMFerreira) [reply aws/aws_instance.go:329]: Please make sure other cloud providers also return this error.

**Review by fabioDMFerreira — CHANGES_REQUESTED** [review]
It requires some minor changes. Looks good to me though.

File: lepton/provider.go (hunk 2 above)
Comment (arknable) [reply lepton/provider.go]: Fixed by https://github.com/nanovms/ops/pull/1034/commits/99369f74d2f8b633ea1b35b374a12ed199e7cc12

File: digitalocean/digital_ocean_instance.go (hunk 1 above)
Comment (arknable) [reply digitalocean/digital_ocean_instance.go]: both styles are fine for me, but the reason was the other methods in that file do not use named return values.

File: aws/aws_instance.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (arknable) [reply aws/aws_instance.go]: Fixed by https://github.com/nanovms/ops/pull/1034/commits/8cccbe5909af1a947a170bc64ae762b3eb7e920f

**Review by fabioDMFerreira — APPROVED** [review]

**Pull request merged**